### PR TITLE
Improve reliability of tests

### PR DESCRIPTION
**Pull request by olivierdagenais**
This will handle the case where it's not _quite_ the same .NET framework
running, plus any difference will be identified by its line number.

### CODE REVIEW

File: Solink.AddIn.Helpers.Test/RestartableAddInGeneratorTest.cs [hunk 1]
```diff
@@ -83,7 +83,41 @@ private static void AssertGeneratedCode(string expected, CodeCompileUnit compile
...
+                checkLine();                                                 //
+                checkLine();                                                 //     Changes to this file may cause incorrect behavior and will be lost if
+                checkLine();                                                 //     the code is regenerated.
+                checkLine();                                                 // </auto-generated>
+                checkLine();                                                 //-----------------
+                string el = er.ReadLine(), al = ar.ReadLine();
+                while (el != null && al != null)
+                {
+                    checkLine();
+                    el = er.ReadLine(); al = ar.ReadLine();
```
Comment (JefStat) [inline Solink.AddIn.Helpers.Test/RestartableAddInGeneratorTest.cs]: `checkLine` and this line call readline. In essence your checking every second line now in this loop. Also line number is not incremented.

File: Solink.AddIn.Helpers.Test/RestartableAddInGeneratorTest.cs [hunk 2]
```diff
@@ -83,7 +83,41 @@ private static void AssertGeneratedCode(string expected, CodeCompileUnit compile
...
+                checkLine();                                                 // <auto-generated>
+                checkLine();                                                 //     This code was generated by a tool.
+                lineNumber++; er.ReadLine(); ar.ReadLine(); // don't care    //     Runtime Version:4.0.30319.34209
+                checkLine();                                                 //
+                checkLine();                                                 //     Changes to this file may cause incorrect behavior and will be lost if
+                checkLine();                                                 //     the code is regenerated.
+                checkLine();                                                 // </auto-generated>
+                checkLine();                                                 //-----------------
+                string el = er.ReadLine(), al = ar.ReadLine();
+                while (el != null && al != null)
```
Comment (JefStat) [inline Solink.AddIn.Helpers.Test/RestartableAddInGeneratorTest.cs]: `||` would be sufficient here and a slightly faster for short circuit evaluation.

File: Solink.AddIn.Helpers.Test/RestartableAddInGeneratorTest.cs [hunk 3]
```diff
@@ -83,7 +83,41 @@ private static void AssertGeneratedCode(string expected, CodeCompileUnit compile
...
+                checkLine();                                                 //-----------------
+                checkLine();                                                 // <auto-generated>
+                checkLine();                                                 //     This code was generated by a tool.
+                lineNumber++; er.ReadLine(); ar.ReadLine(); // don't care    //     Runtime Version:4.0.30319.34209
+                checkLine();                                                 //
+                checkLine();                                                 //     Changes to this file may cause incorrect behavior and will be lost if
+                checkLine();                                                 //     the code is regenerated.
+                checkLine();                                                 // </auto-generated>
+                checkLine();                                                 //-----------------
+                string el = er.ReadLine(), al = ar.ReadLine();
```
Comment (JefStat) [inline Solink.AddIn.Helpers.Test/RestartableAddInGeneratorTest.cs]: Another problem with reading the line and not asserting the values before reading the next line. Also line number is not incremented.

**Comment by olivierdagenais** [comment]
I fixed the embarrassing oversights and added more tests.

**Pull request merged**